src/app/ReadClient.h

```diff
@@ -164,6 +164,13 @@ class ReadClient : public Messaging::ExchangeDelegate
          * @param[in] apReadClient The read client object of the terminated read or subscribe interaction.
          */
         virtual void OnDone(ReadClient * apReadClient) = 0;
+
+        /**
+         * This function is invoked when using SendAutoResubscribeRequest, where the ReadClient was configured to auto re-subscribe and the ReadPrepareParams was
+         * moved into this client for management. This will have to be free'ed appropriately by the application.
+         * OnDeallocatePaths is called in ReadClient's destructor when OnDone is called and delete ReadClient
```

Comment:
OK, so I have several concerns here:

1) This function will get called even if OnDone has not been called.  In fact, it will get called even if SendAutoResubscribeRequest was never called.
2) By the time this function is being called ... what makes sure the callback is alive?  Nothing says the callback has to delete the ReadClient in OnDone before deleting itself.  In particular, some callbacks delete the ReadClient by deleting themselves and having it as a member, which means we'll be calling virtual methods on partially destroyed objects, which is .... not good.

I think what we should implement is the following semantics:

* If SendAutoResubscribeRequest fails, this function will be called before it returns the failure.
* If SendAutoResubscribeRequest succeeds, this function will be called immediately before calling OnDone.
* If SendAutoResubscribeRequest is not called, this function will not be called.  We can detect whether SendAutoResubscribeRequest was called by testing `mReadPrepareParams.mResubscribePolicy != nullptr`, right?

Reply:
Fully agree on it, done, thanks